src/serializer/ResidualFunctions.js

```diff
@@ -472,10 +472,13 @@ export class ResidualFunctions {
           let { functionValue, residualFunctionBindings, insertionPoint } = instance;
           let functionId = this.locationService.getLocation(functionValue);
           invariant(functionId !== undefined);
+          let hasFunctionArg = false;
```

Comment:
At high level, I was surprised that "firstUsage.isNotEarlierThan()" below did not catch this testcase. Because we are essentially use it before the function's insertion point and first usage checking is more accurate.
In future PR we may want to check the residualBinding.value is not serialized yet, which will give us more accurate result.